sa/storage-authority_test.go

```diff
@@ -395,3 +405,55 @@ func TestUpdateOCSP(t *testing.T) {
 	test.AssertNotError(t, err, "Failed to fetch OCSP response")
 	test.AssertEquals(t, ocspResponse, string(fetchedOcspResponse.Response))
 }
+
+func TestMarkCertificateRevoked(t *testing.T) {
+	sa, fc, cleanUp := initSA(t)
+	defer cleanUp()
+
+	reg := satest.CreateWorkingRegistration(t, sa)
+	// Add a cert to the DB to test with.
+	certDER, err := ioutil.ReadFile("www.eff.org.der")
+	test.AssertNotError(t, err, "Couldn't read example cert DER")
+	_, err = sa.AddCertificate(certDER, reg.ID)
+	test.AssertNotError(t, err, "Couldn't add www.eff.org.der")
+
+	serial := "00000000000000000000000000021bd4"
+	const ocspResponse = "this is a fake OCSP response"
+
+	certificateStatusObj, err := sa.dbMap.Get(core.CertificateStatus{}, serial)
+	beforeStatus := certificateStatusObj.(*core.CertificateStatus)
+	test.AssertEquals(t, beforeStatus.Status, core.OCSPStatusGood)
+
+	fc.Add(1 * time.Hour)
+
+	code := core.RevocationCode(1)
+	err = sa.MarkCertificateRevoked(serial, []byte(ocspResponse), code)
+	test.AssertNotError(t, err, "MarkCertificateRevoked failed")
+
+	certificateStatusObj, err = sa.dbMap.Get(core.CertificateStatus{}, serial)
+	afterStatus := certificateStatusObj.(*core.CertificateStatus)
+	test.AssertNotError(t, err, "Failed to fetch certificate status")
+
+	if code != afterStatus.RevokedReason {
+		t.Errorf("RevokedReasons, expected %s, got %s", code, afterStatus.RevokedReason)
+	}
+	if !fc.Now().Equal(afterStatus.RevokedDate) {
+		t.Errorf("RevokedData, expected %s, got %s", fc.Now(), afterStatus.RevokedDate)
+	}
+	if !fc.Now().Equal(afterStatus.OCSPLastUpdated) {
+		t.Errorf("OCSPLastUpdated, expected %s, got %s", fc.Now(), afterStatus.OCSPLastUpdated)
```

Comment:
Your clock makes these tests so much nicer. Why didn't I run into this pattern fourteen years ago?